mavros/src/plugins/setpoint_raw.cpp

```diff
@@ -218,24 +222,44 @@ class SetpointRawPlugin : public plugin::PluginBase,
 	{
 		Eigen::Quaterniond desired_orientation;
 		Eigen::Vector3d baselink_angular_rate;
-
+		Eigen::Vector3d body_rate;
+		double thrust;
+		// Set Thrust scaling in px4_config.yaml, setpoint_raw block.
+		if (!sp_nh.getParam("thrust_scaling", thrust_scaling) && req->thrust != 0.0) {
+			ignore_thrust_message_ = true;
+		}
```

Comment:
One of the problems of getParam - it is not fast, but at least it have cache (in roscpp).
That style will look a bit better:
```cpp
auto ignore_thrust = req->thrust != 0.0 && !sp_nh.getParam(...)
```

Reply:
Usually we get parameter in `initialize()`, but your way allows to tune value.

Reply:
Just checked this now. I will do the way you suggest for the definition, and keep the get_param inside the handle to allow re-tuning. But I agree with you, it's not optimal as it is slow, I might try to optimize it later on...